### ISSUE TITLE
Validate and Load all PDS4 MESSENGER data products with Nucleus

### ISSUE THREAD

**Issue by jordanpadams** [NASA-PDS/nucleus#54]
## 💡 Description

- Document benchmark metrics for execution time, cost, ?

**Comment by tloubrieu-jpl** [comment]
@ramesh-maddegoda  focuses on MSGRMDS_4001 and MESSDEM_1001 that need to be loaded in the registry first so that they can be used in ticket https://github.com/NASA-PDS/search-api-notebook/issues/24

**Comment by tloubrieu-jpl** [comment]
Blocked because AWS Airflow is unavailable on NGAP

**Comment by tloubrieu-jpl** [comment]
Unblocked since Ramesh work on MCP. He is now testing the ECS task called by the nucleus workflow.

**Comment by jordanpadams** [comment]
Status: @ramesh-maddegoda working on improving Terraform deployments

**Comment by tloubrieu-jpl** [comment]
@ramesh-maddegoda is deploying everything needed on MCP, from scratch.

**Comment by tloubrieu-jpl** [comment]
@ramesh-maddegoda will test nucleus to validate its robustness with a bigger dataset.

**Comment by ramesh-maddegoda** [comment]
Some of the files in the s3://asc-pds-messenger failed to copy to the PDS Nucleus staging bucket with a permission issue.

```
aws s3 cp s3://asc-pds-messenger/MSGRMDS_8001/RTM/MDIS_RTM_N01/2013_228/MDIS_RTM_N01_006974_4644396_1.IMG s3://pds-nucleus-staging/messenger-data/MSGRMDS_8001/RTM/MDIS_RTM_N01/2013_228/
```
```
copy failed: s3://asc-pds-messenger/MSGRMDS_8001/RTM/MDIS_RTM_N01/2013_228/MDIS_RTM_N01_006974_4644396_1.IMG to s3://pds-nucleus-staging/messenger-data/MSGRMDS_8001/RTM/MDIS_RTM_N01/2013_228/MDIS_RTM_N01_006974_4644396_1.IMG An error occurred (AccessDenied) when calling the GetObjectTagging operation: Access Denied
```

**Comment by tloubrieu-jpl** [comment]
A new parameter to copy only some of the metadata enable to copy the messenger data.

**Comment by tloubrieu-jpl** [comment]
@ramesh-maddegoda identified a bug while doing that test. The lambda reading the data sync report is now taking more that 15 minutes. Now there will be a single lambda call per report.

**Comment by nutjob4life** [comment]
20,000 processed! 8 directories ran! Found 2 errors:

- "Capacity not available at the moment", related to ECS, seems to have run out; need to have "capacity providers" but cost is a concern; see related bug in Nucleus
- Role error: have seen it before, error occured by reading data file. An error in validate‽ Trying to isolate it with the same Docker image and reproduce it without Nucleus. Is it Airflow or validate?

**Comment by tloubrieu-jpl** [comment]
- capacity issue is solved with a retry configuration.
- validate error is an actual validation error.

**Comment by tloubrieu-jpl** [comment]
@ramesh-maddegoda is experimenting with SQS to send to new records to the mysql database and avoid the time out he was experiencing with direct insertion.

**Comment by tloubrieu-jpl** [comment]
SQS now mostly works, but another lambda had a time out.

**Comment by tloubrieu-jpl** [comment]
We now integrate the copy from S3 to EFS as a nucleus step in the DAGs. We give up DataSync which comes with risks of overlapping copies and complication to remove files from EFS.

@ramesh-maddegoda will also write a note in a wiki for a future design where we don't need to use EFS at all.

**Comment by jordanpadams** [comment]
This work has been paused as we focus on Catalina Sky Survey. Will move to B15.0 release plan to complete work.

**Comment by jordanpadams** [comment]
📆 05/2024 status: Delayed several sprints due to delays in https://github.com/NASA-PDS/nucleus/issues/93. This is an operations activity. No impact on build.

**Comment by jordanpadams** [comment]
📆 06/2024 status: Delayed several sprints due to delays in https://github.com/NASA-PDS/nucleus/issues/93. This is an operations activity. No impact on build.

**Comment by jordanpadams** [comment]
📆 07/2024 status: Delayed several sprints due to delays in https://github.com/NASA-PDS/nucleus/issues/93. This is an operations activity. No impact on build.

**Comment by jordanpadams** [comment]
📆 08/2024 status: Delayed several sprints due to delays in https://github.com/NASA-PDS/nucleus/issues/93. This is an operations activity. No impact on build. Will most likely be deferred to B15.1

**Comment by jordanpadams** [comment]
📆 09/2024 status: deferred to B15.1 due to delays in https://github.com/NASA-PDS/nucleus/issues/93. This is an operations activity. No impact on build.

**Comment by jordanpadams** [comment]
Closing as wontfix. Eventually we will hand off this task to IMG.
